cuebot/scheduler-sim/tag_gpu_watch.py

```
@@ -0,0 +1,179 @@
+NTAGS = int(os.environ.get("SIM_NTAGS", "0") or "0")
+MIN_GPU = int(os.environ.get("SIM_TAGGPU_MIN_GPU", "50"))
+# GPU utilization floor (percent of GPU UNITS on GPU-capable hosts only --
+# most of the farm has no GPUs and must not dilute the denominator).
+# Calibrated: the full-farm scenario measures ~75% peak.
+MIN_UTIL = float(os.environ.get("SIM_TAGGPU_MIN_UTIL", "40"))
+CSV = os.environ.get("SIM_TAGGPU_CSV", "")
+
+
+def _scalar(sql, default=0):
+    try:
+        out = subprocess.run(PSQL + ["-c", sql], capture_output=True, text=True,
```

Ruff (0.16.0)

[error] `subprocess` call: check for execution of untrusted input

(S603)